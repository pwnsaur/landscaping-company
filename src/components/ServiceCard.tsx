import Image from 'next/image';
import Link from 'next/link';
import { TypeService } from 'types';
import styled from 'styled-components';
import ErrorBoundary from '@/utils/ErrorBoundary';

const Service = ({ service }: { service: TypeService }) => {
  const { coverImage, slug, title, description, excerpt } = service.fields;

  return (
    <ErrorBoundary>
      <Break />
      <Card>
        <StyledLink href={`/services/${slug}`}>
          <StyledImage
            src={`https:${coverImage.fields.file.url}`}
            alt='cover-image'
            height={coverImage.fields.file.details.image!.height}
            width={coverImage.fields.file.details.image!.width}
          />
          <Description>
            <Title>{title}</Title>
            <Excerpt>{excerpt}</Excerpt>
          </Description>
        </StyledLink>
      </Card>
    </ErrorBoundary>
  );
};

export default Service;

const Break = styled.hr`
  border: none;
  height: 1px;
  background-color: black;
  margin: 20px 0;
  width: 100%;
`;

const Card = styled.div`
  /* display: flex;
  flex-direction: row; */
  justify-content: center;
  width: 100%;
  /* padding: 20px; */
  margin: 20px auto;
  text-align: center;
  /* background-color: #fff; */
`;

const Title = styled.div`
  font-size: 1.5rem;
  /* margin-bottom: 10px; */
  margin: auto;
`;

const StyledImage = styled(Image)`
  width: 40%;
  height: auto;
`;

const StyledLink = styled(Link)`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  /* width: 1100px; */
`;

const Excerpt = styled.p`
  margin: 10px 20px;
`;

const Description = styled.div`
  width: 100%;
`;
