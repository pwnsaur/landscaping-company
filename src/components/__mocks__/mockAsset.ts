import { Asset } from 'contentful';

export const createMockAsset = (): Asset => {
  return {
    sys: {
      id: '1',
      type: 'Asset',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'asset',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'jonathan-kemper-kYl3Yp2hwtQ-unsplash',
      description: '',
      file: {
        url: '//images.ctfassets.net/sandie-clarke-q13Zq1Jufks-unsplash.jpg',
        details: {
          size: 500939,
          image: {
            width: 1920,
            height: 1080,
          },
        },
        fileName: '',
        contentType: 'image/jpeg',
      },
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Asset),
  };
};
