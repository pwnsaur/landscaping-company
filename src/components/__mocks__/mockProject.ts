import { createMockAsset } from '@components/__mocks__/mockAsset';
import { BLOCKS } from '@contentful/rich-text-types';
import { Entry } from 'contentful';

import { TypeProjectFields } from '@/types/contentfulTypes';

export const createMockProject = (): Entry<TypeProjectFields> => {
  return {
    sys: {
      id: '1',
      type: 'Entry',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'project',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'Fifteenth Project',
      slug: 'fifteenth-project',
      content: {
        nodeType: BLOCKS.DOCUMENT,
        data: {},
        content: [],
      },
      coverImage: createMockAsset(),
      date: '2023-04-27',
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Entry<TypeProjectFields>),
    update: () => Promise.resolve({} as Entry<TypeProjectFields>),
  };
};
