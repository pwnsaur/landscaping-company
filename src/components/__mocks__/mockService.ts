import { createMockAsset } from '@components/__mocks__/mockAsset';
import { BLOCKS } from '@contentful/rich-text-types';
import { Entry } from 'contentful';

import { TypeServiceFields } from '@/types/contentfulTypes';

export const createMockService = (): Entry<TypeServiceFields> => {
  return {
    sys: {
      id: '1',
      type: 'Entry',
      createdAt: '2023-04-27T12:00:00Z',
      updatedAt: '2023-04-27T12:00:00Z',
      locale: 'en-US',
      contentType: {
        sys: {
          id: 'service',
          type: 'Link',
          linkType: 'ContentType',
        },
      },
    },
    fields: {
      title: 'Service',
      slug: 'service',
      excerpt: 'service',
      description: {
        nodeType: BLOCKS.PARAGRAPH,
        data: {},
        content: [],
      },
      coverImage: createMockAsset(),
    },
    metadata: {
      tags: [],
    },
    toPlainObject: () => ({} as Entry<TypeServiceFields>),
    update: () => Promise.resolve({} as Entry<TypeServiceFields>),
  };
};
