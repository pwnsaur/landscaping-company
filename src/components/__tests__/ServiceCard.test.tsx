import React from 'react';
import { render } from '@testing-library/react';
import { createMockService } from '@/components/__mocks__/mockService';
import ServiceCard from '@/components/ServiceCard';

const mockService = createMockService();

describe('ServiceCard', () => {
  test('renders correctly', () => {
    const { getByText, asFragment } = render(
      <ServiceCard service={mockService} priority={false} />
    );

    expect(getByText(mockService.fields.title)).toBeInTheDocument();

    if (mockService.fields.excerpt) {
      expect(getByText(mockService.fields.excerpt)).toBeInTheDocument();
    }

    expect(asFragment()).toMatchSnapshot();
  });
});
