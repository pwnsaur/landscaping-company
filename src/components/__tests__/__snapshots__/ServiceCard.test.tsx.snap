// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  height: 1px;
  background-color: black;
  margin: 1rem 0;
  width: 100%;
}

<hr
    class="c0"
  />
  .c0 {
  display: flex;
  justify-content: center;
  width: 100%;
  margin: 0 auto;
  text-align: center;
}

.c4 {
  font-size: clamp(1.2rem, 2vw, 2rem);
  font-weight: 400;
  margin: auto;
}

.c2 {
  max-width: 40%;
  height: auto;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
}

.c5 {
  margin: 10px 20px;
  font-size: clamp(0.8rem, 1.5vw, 1.2rem);
}

.c3 {
  width: 100%;
}

<div
    class="c0"
  >
    <a
      class="c1"
      href="/services/service"
    >
      <img
        alt="cover-image"
        class="c2"
        data-nimg="1"
        decoding="async"
        height="270"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=1080&q=50"
        srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=640&q=50 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Fsandie-clarke-q13Zq1Jufks-unsplash.jpg&w=1080&q=50 2x"
        style="color: transparent;"
        width="480"
      />
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          Service
        </h2>
        <p
          class="c5"
        >
          service
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
